tests/system/Helpers/FormHelperTest.php

```diff
@@ -52,25 +52,40 @@ private function setRequest(): void
         Services::injectMock('request', $request);
     }
 
+    private function setCsrfFilter(): void
+    {
+        $filters                      = config(Filters::class);
+        $filters->globals['before'][] = 'csrf';
+        service('filters')->initialize();
+    }
+
     public function testFormOpenBasic(): void
     {
         $this->setRequest();
 
-        $before = (new Filters())->globals['before'];
-        if (in_array('csrf', $before, true) || array_key_exists('csrf', $before)) {
-            $Value    = csrf_hash();
-            $Name     = csrf_token();
-            $expected = <<<EOH
-                <form action="http://example.com/index.php/foo/bar" name="form" id="form" method="POST" accept-charset="utf-8">
-                <input type="hidden" name="{$Name}" value="{$Value}" style="display:none;">
+        $expected = <<<'EOH'
+            <form action="http://example.com/index.php/foo/bar" name="form" id="form" method="POST" accept-charset="utf-8">
+
+            EOH;
+        $attributes = [
+            'name'   => 'form',
+            'id'     => 'form',
+            'method' => 'POST',
+        ];
+        $this->assertSame($expected, form_open('foo/bar', $attributes));
+    }
 
-                EOH;
-        } else {
-            $expected = <<<'EOH'
-                <form action="http://example.com/index.php/foo/bar" name="form" id="form" method="POST" accept-charset="utf-8">
+    public function testFormOpenBasicWithCsrf(): void
+    {
+        $this->setRequest();
+        $this->setCsrfFilter();
 
-                EOH;
-        }
+        $Value    = csrf_hash();
+        $Name     = csrf_token();
```

Comment:
While you're here, can you change these variables to lowercase? These are unsightly to look at.

Reply:
Okay. Done.